routes/api.js

```diff
@@ -238,58 +240,99 @@ var removeSession = function(sessionId) {
     return delete sessions[sessionId];
 };
 
-var proxyRoute = function(req, res, next) {
-    proxy.web(req, res);
+// Create a kernel. Does not use the proxy instance because it must parse the
+// and reserialize the request body with additional information in certain
+// configurations.
+router.post('/kernels', bodyParser.json({ type: 'text/plain' }), function(req, res) {
```

Comment:
This version also has the benefit of having all the magic stuff that needs to happen when creating new kernels all in one spot.